app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -10,110 +10,235 @@ import androidx.compose.foundation.layout.Spacer
 import androidx.navigation.NavController
+import com.example.tasky.R
+import com.example.tasky.feature_agenda.domain.model.AgendaItem
+import com.example.tasky.feature_agenda.domain.util.AgendaItemKey
+import com.example.tasky.feature_agenda.domain.util.toAgendaItemType
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.AgendaItemCard
 import com.example.tasky.feature_agenda.presentation.agenda_screen.components.DatePickerDialog
 import com.example.tasky.feature_agenda.presentation.agenda_screen.components.DayChip
 import com.example.tasky.feature_agenda.presentation.agenda_screen.components.TopSection
+import com.example.tasky.feature_agenda.presentation.util.formatDateTimeOfPattern
 import com.example.tasky.presentation.theme.BackgroundBlack
 import com.example.tasky.presentation.theme.BackgroundWhite
+import com.example.tasky.util.ObserveAsEvents
 import com.example.tasky.util.Result
 import com.example.tasky.util.Screen
-import com.vanpra.composematerialdialogs.rememberMaterialDialogState
+import com.vanpra.composematerialdialogs.MaterialDialogState
+import kotlinx.coroutines.flow.Flow
 import java.time.ZoneId
-import java.time.format.DateTimeFormatter
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun AgendaScreen(
     navController: NavController,
-    viewModel: AgendaViewModel = hiltViewModel()
+    state: AgendaState,
+    onEvent: (AgendaEvent) -> Unit,
+    dateDialogState: MaterialDialogState,
+    username: String,
+    logoutEventsChannelFlow: Flow<Result<Unit>>
 ) {
-    viewModel.dateDialogState = rememberMaterialDialogState()
-    val state by viewModel.state.collectAsStateWithLifecycle()
-
     val context = LocalContext.current
 
-    LaunchedEffect(key1 = true) {
-        viewModel.logoutResult.collect { result ->
-            when(result) {
-                is Result.Error -> Toast.makeText(context, result.message, Toast.LENGTH_LONG).show()
-                is Result.Success -> navController.navigate(Screen.LoginScreen.route)
-            }
+    ObserveAsEvents(logoutEventsChannelFlow) { result ->
+        when(result) {
+            is Result.Error -> Toast.makeText(context, result.message, Toast.LENGTH_LONG).show()
+            is Result.Success -> navController.navigate(Screen.LoginScreen.route)
         }
     }
 
-    Column(
-        modifier = Modifier
-            .fillMaxSize()
-            .background(BackgroundBlack)
-    ) {
-        TopSection(
-            state = state,
-            dateDialogState = viewModel.dateDialogState,
-            username = viewModel.username,
-           showMenuOptions = {
-                viewModel.onEvent(AgendaEvent.ToggleLogoutBtn(it))
-            },
-            logout = {
-                viewModel.onEvent(AgendaEvent.Logout)
+    Scaffold(
+        floatingActionButton = {
+            FloatingActionButton(
+                onClick = {
+                    onEvent(AgendaEvent.ToggleItemCreationMenu(!state.isItemCreationMenuVisible))
```

Comment:
Passing the new value shouldn't be necessary when toggling it. The VM already knows the old one and can therefore just negate it :)